### PR TITLE
[Enhancement] Add FORWARD_NO_SYNC for all SHOW PROC statements

### LINKED ISSUE #3884: [Enhancement] Add FORWARD_NO_SYNC for all SHOW PROC statements
Now some SHOW PROC statementss have no message when executed
in follower. These message are only in master.

### PR DESCRIPTION
**Pull request by chaoyli**
Now some SHOW PROC statementss have no message when executed
in follower. These message are only in master. So I add
FORWARD_NO_SYNC for all SHOW PROC statements.

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/analysis/ShowProcStmt.java [hunk 1]
```diff
@@ -82,10 +82,6 @@ public ShowResultSetMetaData getMetaData() {
 
     @Override
     public RedirectStatus getRedirectStatus() {
```
Comment (kangkaisen) [inline fe/src/main/java/org/apache/doris/analysis/ShowProcStmt.java:84]: I think rename the `forward_to_master` to `show_stmt_forward_to_master`, and change the default value to true is a better way.

**All show stmt should have the consistent behavior.**

**Comment by morningman** [comment]
Hi @chaoyli . Actually, not all `ShowProc` stmt should be forward to master.
Some of them, yes, like: `show proc "/cluster_balance"`, because the non-master FE has
nothing to do the the cluster balance logic.

But some of them don't, like: `show proc "/dbs/10003/10017/partitions/10016/10018";`. 
This is the stmt to show the tablet info saving in the current FE's memory.

These kind of `show proc` stmt should not be forward to master FE because sometimes
we need to check the meta of current FE, to see if there is any inconsistency with the Master FE.

That is why I add the variable `forward_to_master` to control whether to forward these stmts.

**Comment by morningman** [comment]
And by the way, better to fix the compilation problem in another PR.
That is more urgent and need to be fixed and merged ASAP.

**Comment by chaoyli** [comment]
> Hi @chaoyli . Actually, not all `ShowProc` stmt should be forward to master.
> Some of them, yes, like: `show proc "/cluster_balance"`, because the non-master FE has
> nothing to do the the cluster balance logic.
> 
> But some of them don't, like: `show proc "/dbs/10003/10017/partitions/10016/10018";`.
> This is the stmt to show the tablet info saving in the current FE's memory.
> 
> These kind of `show proc` stmt should not be forward to master FE because sometimes
> we need to check the meta of current FE, to see if there is any inconsistency with the Master FE.
> 
> That is why I add the variable `forward_to_master` to control whether to forward these stmts.

**Comment by chaoyli** [comment]
> Hi @chaoyli . Actually, not all `ShowProc` stmt should be forward to master.
> Some of them, yes, like: `show proc "/cluster_balance"`, because the non-master FE has
> nothing to do the the cluster balance logic.
> 
> But some of them don't, like: `show proc "/dbs/10003/10017/partitions/10016/10018";`.
> This is the stmt to show the tablet info saving in the current FE's memory.
> 
> These kind of `show proc` stmt should not be forward to master FE because sometimes
> we need to check the meta of current FE, to see if there is any inconsistency with the Master FE.
> 
> That is why I add the variable `forward_to_master` to control whether to forward these stmts.

I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And I don't know whether to set the variable or not. The user must remember every show statement which it's semantics.

**Comment by morningman** [comment]
> I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.

It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.

**Comment by chaoyli** [comment]
> And by the way, better to fix the compilation problem in another PR.
> That is more urgent and need to be fixed and merged ASAP.

It's because I run the unit test before I push the PR, this is by the way.

**Comment by chaoyli** [comment]
> > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> 
> It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.

So I only make show proc get the forward variable to make a comprise.

**Comment by morningman** [comment]
> > > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> > 
> > 
> > It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.
> 
> So I only make show proc get the forward variable to make a comprise.

How to do with this?
`show proc "/dbs/10003/10017/partitions/10016/10018";`

**Comment by chaoyli** [comment]
> > > > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> > > 
> > > 
> > > It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.
> > 
> > 
> > So I only make show proc get the forward variable to make a comprise.
> 
> How to do with this?
> `show proc "/dbs/10003/10017/partitions/10016/10018";`
This is a question. If we make the behavior consistent with master to make balance statistics exists in follower like table metadata.

**Comment by chaoyli** [comment]
> > > > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> > > 
> > > 
> > > It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.
> > 
> > 
> > So I only make show proc get the forward variable to make a comprise.
> 
> How to do with this?
> `show proc "/dbs/10003/10017/partitions/10016/10018";`

This is a question. If we make the behavior consistent with master to make balance statistics exists in follower like table metadata.

**Comment by morningman** [comment]
> > > > > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> > > > 
> > > > 
> > > > It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.
> > > 
> > > 
> > > So I only make show proc get the forward variable to make a comprise.
> > 
> > 
> > How to do with this?
> > `show proc "/dbs/10003/10017/partitions/10016/10018";`
> 
> This is a question. If we make the behavior consistent with master to make balance statistics exists in follower like table metadata.

That could cause too many meta sync log.
Maybe we can find a way to classify the `show proc` statement. This can control the behavior of each category.

**Comment by chaoyli** [comment]
> > > > > > I thinks the variable forward_to_master is useless, even make application complicated. User deploy the Doris will one Master and two Follower. The master will change when new vote. And the user don't know whether to set the variable or not. He must remember every show statement implementation difference.
> > > > > 
> > > > > 
> > > > > It indeed complicated. But I can't figure out how to make it simple for user to see both master and non-master FE's meta info.
> > > > 
> > > > 
> > > > So I only make show proc get the forward variable to make a comprise.
> > > 
> > > 
> > > How to do with this?
> > > `show proc "/dbs/10003/10017/partitions/10016/10018";`
> > 
> > 
> > This is a question. If we make the behavior consistent with master to make balance statistics exists in follower like table metadata.
> 
> That could cause too many meta sync log.
> Maybe we can find a way to classify the `show proc` statement. This can control the behavior of each category.

OK, we should found another method to address the problem.

**Pull request closed without merging**